File: native/android/VueNativeCore/src/main/kotlin/com/vuenative/core/Components/Factories/VProgressBarFactory.kt
package com.vuenative.core

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.ProgressBar

class VProgressBarFactory : NativeComponentFactory {
    override fun createView(context: Context): View {
        return ProgressBar(context, null, android.R.attr.progressBarStyleHorizontal).apply {
            isIndeterminate = false
            max = 1000
            progress = 0
            layoutParams = ViewGroup.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.WRAP_CONTENT
            )
        }
    }

    override fun updateProp(view: View, key: String, value: Any?) {
        val pb = view as? ProgressBar ?: return
        when (key) {
            "progress" -> {
                val f = StyleEngine.toFloat(value, 0f).coerceIn(0f, 1f)
                pb.progress = (f * pb.max).toInt()
            }
            "progressTintColor" -> {
                val color = StyleEngine.parseColor(value) ?: return
                pb.progressTintList = android.content.res.ColorStateList.valueOf(color)
            }
            "trackTintColor" -> {
                val color = StyleEngine.parseColor(value) ?: return
                pb.progressBackgroundTintList = android.content.res.ColorStateList.valueOf(color)
            }
            "animated" -> {} // Android ProgressBar animates by default
            else -> StyleEngine.apply(key, value, view)
        }
    }

    override fun addEventListener(view: View, event: String, handler: (Any?) -> Unit) {}
    override fun removeEventListener(view: View, event: String) {}
}

File: native/ios/VueNativeCore/Sources/VueNativeCore/Components/Factories/VProgressBarFactory.swift
#if canImport(UIKit)
import UIKit
import FlexLayout

/// Factory for VProgressBar — maps to UIProgressView.
final class VProgressBarFactory: NativeComponentFactory {

    func createView() -> UIView {
        let bar = UIProgressView(progressViewStyle: .default)
        bar.progress = 0
        _ = bar.flex
        return bar
    }

    func updateProp(view: UIView, key: String, value: Any?) {
        guard let bar = view as? UIProgressView else {
            StyleEngine.apply(key: key, value: value, to: view)
            return
        }
        switch key {
        case "progress":
            let p = (value as? Double) ?? (value as? NSNumber)?.doubleValue ?? 0
            bar.setProgress(Float(max(0, min(1, p))), animated: false)
        case "progressTintColor":
            if let str = value as? String { bar.progressTintColor = UIColor.fromHex(str) }
        case "trackTintColor":
            if let str = value as? String { bar.trackTintColor = UIColor.fromHex(str) }
        case "animated":
            // stored for use in future progress updates — no-op here
            break
        default:
            StyleEngine.apply(key: key, value: value, to: view)
        }
    }

    func addEventListener(view: UIView, event: String, handler: @escaping (Any?) -> Void) {}
    func removeEventListener(view: UIView, event: String) {}
}
#endif
